plugins/Hosting/js/siaMath.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+// Library for arbitrary precision in numbers
+const BigNumber = require('bignumber.js');
+// Ensure precision
+BigNumber.config({ DECIMAL_PLACES: 24 });
+BigNumber.config({ EXPONENTIAL_AT: 1e+9 });
+// Display # of decimals
+const DISPLAY_DECIMALS = 2;
+
+// Catch if BigNumber, property object, or anything else is passed in
+function interpret(thing) {
+	if (typeof thing === 'object') {
+		thing = thing.value;
+	}
+	return new BigNumber(thing).toNumber();
+}
```

Comment:
It's important to read them with the context of hostData.js, But I'll try to make the file more explicit